discojs/discojs-core/src/training/disco.ts

```diff
@@ -1,17 +1,18 @@
-import type { data, Logger, Memory, Task, TrainingInformant, TrainingInformation } from '../index.js'
-import { client as clients, informant as informants, EmptyMemory, ConsoleLogger } from '../index.js'
-import type { Trainer } from './trainer/trainer.js'
-import { TrainerBuilder } from './trainer/trainer_builder.js'
-import type { TrainerLog } from '../logging/trainer_logger.js'
+import { List } from 'immutable'
+
+import type { data, Logger, Memory, Task, TrainingInformation } from '../index.js'
+import { client as clients, EmptyMemory, ConsoleLogger } from '../index.js'
 import type { Aggregator } from '../aggregator/index.js'
 import { MeanAggregator } from '../aggregator/mean.js'
 
+import type { RoundLogs, Trainer } from './trainer/trainer.js'
+import { TrainerBuilder } from './trainer/trainer_builder.js'
+
 export interface DiscoOptions {
   client?: clients.Client
   aggregator?: Aggregator
   url?: string | URL
   scheme?: TrainingInformation['scheme']
-  informant?: TrainingInformant
```

Comment:
I thought the training informant was used to keep track of training metrics, can it be removed because there didn't used to be training logs?

Reply:
This is answered in your lats comment, then can't we also use the logs for the graph informant? I'm not sure what's the difference between training informant and graph informant

Reply:
> I thought the training informant was used to keep track of training metrics

indeed, it was, but was partially done and IMO a bad design: there is no clear information on when the metrics are valids, when to reset theses, what was exposed (mutating state is hard). having a generator of logs is way safer and also clearer as when values are generated.

> I'm not sure what's the difference between training informant and graph informant

it's not really clear for me either. from what I gathered, it is used for keeping a list of recently generated logs of training in order to build an ApexChart. it's is only used in `Validator` now but I didn't rework the latter so I'm kind of scared of digging another rabbithole with wanting to remove it whole. anyway, let's keep it for now.